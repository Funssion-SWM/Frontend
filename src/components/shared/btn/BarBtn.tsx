type Props = {
  text: string;
  onClick: () => void;
  isSelected: boolean;
};

export default function BarBtn({ text, onClick, isSelected }: Props) {
  return (
    <button
      className={`flex-1 font-semibold py-1 transition-all ${
        isSelected
          ? 'text-soma-blue-50 border-b-[2px] border-soma-blue-50 '
          : 'text-soma-grey-50 hover:text-soma-blue-50 transition'
      }`}
      onClick={() => onClick()}
    >
      {text}
    </button>
  );
}
