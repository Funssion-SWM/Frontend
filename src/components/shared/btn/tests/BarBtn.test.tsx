import { render, screen } from '@testing-library/react';
import BarBtn from '../BarBtn';
import renderer from 'react-test-renderer';
import userEvent from '@testing-library/user-event';

describe('BatBtn', () => {
  it('renders correctly when isSelected is true', () => {
    const component = renderer.create(
      <BarBtn text="text" onClick={() => {}} isSelected={true} />
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders correctly when isSelected is false', () => {
    const component = renderer.create(
      <BarBtn text="text" onClick={() => {}} isSelected={false} />
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('when button clicked', async () => {
    const mockCallBack = jest.fn();

    render(<BarBtn text="text" onClick={mockCallBack} isSelected={true} />);

    const btn = screen.getByRole('button');

    await userEvent.click(btn);
    expect(mockCallBack.mock.calls.length).toEqual(1);
  });
});
