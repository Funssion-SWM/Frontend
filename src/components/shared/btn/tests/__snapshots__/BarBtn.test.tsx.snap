// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BatBtn renders correctly when isSelected is false 1`] = `
<button
  className="flex-1 font-semibold py-1 text-soma-grey-50 hover:text-soma-blue-50 transition"
  onClick={[Function]}
>
  text
</button>
`;

exports[`BatBtn renders correctly when isSelected is true 1`] = `
<button
  className="flex-1 font-semibold py-1 text-soma-blue-50 border-b-[3px] border-soma-blue-50 "
  onClick={[Function]}
>
  text
</button>
`;
